File: src/models/notebook.ts
import type { NotebookAbsence } from "./notebook-absence";
import type { NotebookDelay } from "./notebook-delay";
import type { NotebookObservation } from "./notebook-observation";
import type { NotebookPrecautionaryMeasure } from "./notebook-precautionary-measure";
import type { NotebookPunishment } from "./notebook-punishment";

export type Notebook = Readonly<{
  absences: NotebookAbsence[];
  delays: NotebookDelay[]
  punishments: NotebookPunishment[]
  observations: NotebookObservation[]
  precautionaryMeasures: NotebookPrecautionaryMeasure[]
}>;

File: library/src/commonMain/kotlin/models/Notebook.kt
package models

data class Notebook(
    val absences: List<NotebookAbsence>,
    val delays: List<NotebookDelay>,
    val punishments: List<NotebookPunishment>,
    val observations: List<NotebookObservation>,
    val precautionaryMeasures: List<NotebookPrecautionaryMeasure>
)